packages/webapp/src/components/NotificationToast/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+
+import toast, { Toaster } from 'react-hot-toast';
+import { BidTx, removePendingBidTx, removePendingSettleTx, resetPendingSettleTx, SettleTx } from '../../state/slices/mempool';
+import { closeEthereumMempoolSocket, openEthereumMempoolSocket } from '../../middleware/alchemyMempoolWebsocket';
+import MempoolToast from '../MempoolToast';
+
+const NotificationToast: React.FC<{}> = props => {
+  const dispatch = useAppDispatch();
+
+  const activeAuction = useAppSelector(state => state.auction.activeAuction);
+  const closeToAuctionEnd = useAppSelector(state => state.auction.closeToEnd);
+  const pendingSettleTxs = useAppSelector(state => state.mempool.pendingSettleTxs);
+  const pendingBidTxs = useAppSelector(state => state.mempool.pendingBidTxs);
+  const listeningMempool = useAppSelector(state => state.mempool.listening);
+  const prevSettledBlockHash = useAppSelector(state => state.settlement.prevSettledBlockHash);
+  const blockhash = useAppSelector(state => state.block.blockHash);
+
+  const showToast = (tx: SettleTx | BidTx, time: number) => {
+    toast.custom(
+      <MempoolToast tx={tx} />,
+      {
+        position: "bottom-center",
+        duration: time
+      }
+    )
+  }
+
+  useEffect(() => {
+    pendingSettleTxs.forEach(tx => {
+      showToast(tx, 20000)
+      dispatch(removePendingSettleTx({ hash: tx.hash }));
+    })
+  }, [pendingSettleTxs, dispatch]);
+
+  useEffect(() => {
+    pendingBidTxs.forEach(tx => {
+      if (closeToAuctionEnd) {
+        showToast(tx, 6000)
+      }
+      dispatch(removePendingBidTx({ hash: tx.hash }));
+    })
+  }, [pendingBidTxs, closeToAuctionEnd, dispatch]);
+
+  useEffect(() => {
+    if ((activeAuction === false || closeToAuctionEnd) && !listeningMempool) {
+      dispatch(openEthereumMempoolSocket())
```

Comment:
@nikitattt this is probably the only question that matters in my review. I recently looked through Alchemy's pricing and sockets and mempool subscriptions are super cheap. I think it's better to have all socket work done in a single place, and to avoid complexity, mempool listening can be activated on page load.

Reply:
Thanks for your comment! I have been looking with a fresh eye on it, and indeed `alchemyMempoolSocket` and `alchemySocket` turned out to be nearly the same in the end. I had started with a different idea on how to make it, but it ultimately led to the same thing. I will merge them into one middleware!